kibicara/platforms/email/webapi.py

```diff
@@ -0,0 +1,177 @@
+async def email_create(hood=Depends(get_hood)):
+    """ Create an Email bot. Call this when creating a hood.
+
+    :param hood: Hood row of the hood the Email bot is supposed to belong to.
+    :return: Email row of the new email bot.
+    """
+    try:
+        email_row = await Email.objects.create(hood=hood, secret=urandom(32).hex())
+        spawner.start(email_row)
+        return email_row
+    except IntegrityError:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT)
+
+
+@router.delete('/{email_id}', status_code=status.HTTP_200_OK)
+async def email_delete(email_id, hood=Depends(get_hood)):
+    """ Delete an Email bot. Call this when deleting a hood.
+    Stops and deletes the Email bot as well as all subscribers.
+
+    :param hood: Hood the Email bot belongs to.
+    """
+    email_row = await get_email(hood, email_id=email_id)
+    spawner.stop(email_row)
+    await EmailSubscribers.objects.delete_many(hood=hood.id)
+    await email_row.delete()
+
+
+@router.post('/subscribe/', status_code=status.HTTP_200_OK)
+async def email_subscribe(subscriber: Subscriber, hood=Depends(get_hood_unauthorized)):
+    """ Send a confirmation mail to subscribe to messages via email.
+
+    :param subscriber: Subscriber object, holds the email address.
+    :param hood: Hood the Email bot belongs to.
+    :return: Returns status code 200 after sending confirmation email.
+    """
+    token = to_token(email=subscriber.email)
+    confirm_link = (
+        config['root_url'] + "api/" + str(hood.id) + "/email/subscribe/confirm/" + token
+    )
+    try:
+        send_email(
+            subscriber.email,
+            "Subscribe to Kibicara " + hood.name,
+            sender=hood.name,
+            body="To confirm your subscription, follow this link: " + confirm_link,
+        )
+    except ConnectionRefusedError:
+        logger.info(token)
+        logger.error("Sending subscription confirmation email failed.", exc_info=True)
+        raise HTTPException(status_code=status.HTTP_502_BAD_GATEWAY)
+    except SMTPException:
+        logger.info(token)
+        logger.error("Sending subscription confirmation email failed.", exc_info=True)
+        raise HTTPException(status_code=status.HTTP_502_BAD_GATEWAY)
+
+
+@router.get('/subscribe/confirm/{token}', status_code=status.HTTP_201_CREATED)
+async def email_subscribe_confirm(token, hood=Depends(get_hood_unauthorized)):
+    """ Confirm a new subscriber and add them to the database.
+
+    :param token: encrypted JSON token, holds the email of the subscriber.
+    :param hood: Hood the Email bot belongs to.
+    :return: Returns status code 200 after adding the subscriber to the database.
+    """
+    payload = from_token(token)
+    try:
+        await EmailSubscribers.objects.create(hood=hood.id, email=payload['email'])
+    except IntegrityError:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT)
+
+
+@router.get('/unsubscribe/{token}', status_code=status.HTTP_200_OK)
```

Comment:
should be DELETE, response code 204_NO_CONTENT

Reply:
The response code makes sense, but as it's opened in a browser, I'd use a GET request.

Reply:
Users will eventually get a link that points to a frontend site. The API should use methods with the right semantic.